xarray/test/test_dataset.py

```diff
@@ -1949,6 +1949,46 @@ def test_where(self):
         actual = ds.groupby('c').where(cond)
         self.assertDatasetIdentical(expected, actual)
 
+    def test_sel_where(self):
+
+        # 1d
+        # data array case
```

Comment:
can we add a test case with more interesting coordinates? i.e., something that would broken your previous code where you used `sel` mistakenly instead of `isel`.